### PR TITLE
[#8787] New instructors: set the time zone of the sample course to match user

### LINKED ISSUE #8787: New instructors: set the time zone of the sample course to match user
Steps:
1. Admin creates a new instructor
2. New instructor joins TEAMMATES
3. Upon joining, new instructor is given a sample course 

Current: sample course is in UTC timezone
Suggested: set the timezone to user's time zone

Reason: Sample curse is used to create a first impression, it's nice if the time zone is set correctly.

### PR DESCRIPTION
**Pull request by tshradheya**
Fixes #8787

**Outline of Solution**

- Updates the ZoneId for sample courses and the feedback sessions belonging to that course when a new instructor is joining for the first time.
- Condition `instructor !=null` is to ensure that only the timezone of sample course is updated for instructors joining through the process when the admin has added them
- Since all sessions of a course need to have same timezone so the sample courses and sessions now have `UTC` by default (`InstructorSampleData.json`), causing some godmode changes https://github.com/TEAMMATES/teammates/pull/8851/commits/624b61d4adf89055969426d03e6bac9f5b6fd3e8
- Current test changes so that tests pass irrespective of where they are run. https://github.com/TEAMMATES/teammates/pull/8851/commits/abacbc0dbe5a43ce096b851601fa642957679c68
- Added a new test case in `InstructorCourseJoinAuthenticatedActionTest`

#### How client's timezone is found? 

When the instructor clicks on join link and logins using his google account, then the following page is shown with the `href = "/page/instructorCourseJoinAuthenticated?key=A9CE842BE6AB037AE3F4F9E1992C4E5CBA9E10984E9EAAB79E3754D924122AD28F63413D220AAA2C2D7AB0915CDF4B1A&amp;instructorinstitution=NUS"`
screen shot 2018-06-18 at 2 45 22 pm

Then the Javascript (`instructorCourseJoinConfirmation.js`) run on client adds the `&instructortimezone=Asia/Singapore` to the link which can be then accessed in `InstructorCourseJoinAuthenticatedAction` to change the timezone of sample course and sessions.

### CODE REVIEW

**Comment by tshradheya** [comment]
@darrenwee I have modified the current test and even added a new test case. 

Ready for review

**Comment by tshradheya** [comment]
@darrenwee Reminder for review

File: src/test/java/teammates/test/cases/action/InstructorCourseJoinActionTest.java [hunk 1]
```diff
@@ -25,7 +25,6 @@ protected String getActionUri() {
         return Const.ActionURIs.INSTRUCTOR_COURSE_JOIN;
     }
 
-    @SuppressWarnings("deprecation")
```
[the remark this answers is not in the document]
Comment (tshradheya) [reply src/test/java/teammates/test/cases/action/InstructorCourseJoinActionTest.java]: A function that was previously deprecated was being used here. But in some PR even though it was no longer deprecated, the suppression wasnt removed. Hence I did it here. I can revert this change if you feel its unnecessary for this PR

**Comment by darrenwee** [comment]
Can rebase onto `master`?

**Pull request closed without merging**